Automattic/wp-calypso
# Rename calypso products types

### Changes proposed in this Pull Request

This PR renames some types defined in the `calypso-product` package, to have a cohesive system.

### Implementation notes

- Renamed `JetpackLegacyPlanSlugs` to `JetpackLegacyPlanSlug`
- Renamed `JetpackResetPlanSlugs` to `JetpackResetPlanSlug`
- Renamed `JetpackPlanSlugs` to `JetpackPlanSlug`

### Testing instructions

- Review code
- Check that all checks pass
- Download the PR and boot up Calypso and cloud successfully
